### PR TITLE
Feature/some small things

### LINKED ISSUE #2245: Suggestion: Show building that is currently being worked on
Minecolonies 1.12.2
Show the building that is being built when listing out the required resources at the builder's hut. This saves time when you want to verify if you are building the right building without going to the townhall if you don't need to.

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #2245

# Changes proposed in this pull request:
- Fixes server crash because of sideOnly
- Adds info to the builder GUI of what he works on and where this is.

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/colony/ColonyManager.java [hunk 1]
```diff
@@ -35,6 +35,7 @@
 import net.minecraftforge.common.util.Constants.NBT;
 import net.minecraftforge.fml.common.gameevent.TickEvent;
 import net.minecraftforge.fml.common.network.simpleimpl.IMessage;
+import net.minecraftforge.fml.common.network.simpleimpl.SimpleChannelHandlerWrapper;
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/colony/ColonyManager.java]: MINOR Remove this unused import 'net.minecraftforge.fml.common.network.simpleimpl.SimpleChannelHandlerWrapper'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 63 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 57 major
* ![MINOR][MINOR] 5 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [ColonyManager.java#L957](https://github.com/Minecolonies/minecolonies/blob/141af74be39234859576655799953fdf0f01bd5e/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L957): Refactor this code to not nest more than 5 if/for/while/switch/try statements. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS134)
1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/141af74be39234859576655799953fdf0f01bd5e/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/141af74be39234859576655799953fdf0f01bd5e/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/141af74be39234859576655799953fdf0f01bd5e/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [colony#](https://github.com/Minecolonies/minecolonies/blob/141af74be39234859576655799953fdf0f01bd5e/src/main/java/com/minecolonies/coremod/colony#): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [ColonyManager.java#L56](https://github.com/Minecolonies/minecolonies/blob/141af74be39234859576655799953fdf0f01bd5e/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L56): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 38 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [ColonyManager.java#L56](https://github.com/Minecolonies/minecolonies/blob/141af74be39234859576655799953fdf0f01bd5e/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L56): class "ColonyManager" has 57 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [ColonyManager.java#L332](https://github.com/Minecolonies/minecolonies/blob/141af74be39234859576655799953fdf0f01bd5e/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L332): Provide multiple methods instead of using "canDestroy" to determine which action to take. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2301)
1. ![MAJOR][MAJOR] [ColonyManager.java#L597](https://github.com/Minecolonies/minecolonies/blob/141af74be39234859576655799953fdf0f01bd5e/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L597): Refactor this method to reduce its Cognitive Complexity from 21 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [ColonyManager.java#L929](https://github.com/Minecolonies/minecolonies/blob/141af74be39234859576655799953fdf0f01bd5e/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L929): Refactor this method to reduce its Cognitive Complexity from 31 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/com/minecolonies/coremod/client/gui/WindowBarracksBuilding.java [hunk 2]
```diff
@@ -1,16 +1,66 @@
...
 import com.minecolonies.api.util.constant.Constants;
+import com.minecolonies.blockout.Pane;
+import com.minecolonies.blockout.controls.Label;
+import com.minecolonies.blockout.views.ScrollingList;
+import com.minecolonies.coremod.colony.ColonyView;
 import com.minecolonies.coremod.colony.buildings.BuildingBarracks;
+import net.minecraft.util.math.BlockPos;
 import org.jetbrains.annotations.NotNull;
 
+import java.util.Collections;
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowBarracksBuilding.java]: MINOR Remove this unused import 'java.util.Collections'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 69 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 62 major
* ![MINOR][MINOR] 6 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [ColonyManager.java#L957](https://github.com/Minecolonies/minecolonies/blob/07be2267c2d73fd52859329b09b7bd5d1615f0be/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L957): Refactor this code to not nest more than 5 if/for/while/switch/try statements. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS134)
1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/07be2267c2d73fd52859329b09b7bd5d1615f0be/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/07be2267c2d73fd52859329b09b7bd5d1615f0be/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [WindowHutFarmer.java#L214](https://github.com/Minecolonies/minecolonies/blob/07be2267c2d73fd52859329b09b7bd5d1615f0be/src/main/java/com/minecolonies/coremod/client/gui/WindowHutFarmer.java#L214): Reduce this anonymous class number of lines from 43 to at most 40, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/07be2267c2d73fd52859329b09b7bd5d1615f0be/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [colony#](https://github.com/Minecolonies/minecolonies/blob/07be2267c2d73fd52859329b09b7bd5d1615f0be/src/main/java/com/minecolonies/coremod/colony#): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [ColonyManager.java#L56](https://github.com/Minecolonies/minecolonies/blob/07be2267c2d73fd52859329b09b7bd5d1615f0be/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L56): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 38 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [ColonyManager.java#L56](https://github.com/Minecolonies/minecolonies/blob/07be2267c2d73fd52859329b09b7bd5d1615f0be/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L56): class "ColonyManager" has 57 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [ColonyManager.java#L332](https://github.com/Minecolonies/minecolonies/blob/07be2267c2d73fd52859329b09b7bd5d1615f0be/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L332): Provide multiple methods instead of using "canDestroy" to determine which action to take. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2301)
1. ![MAJOR][MAJOR] [ColonyManager.java#L597](https://github.com/Minecolonies/minecolonies/blob/07be2267c2d73fd52859329b09b7bd5d1615f0be/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L597): Refactor this method to reduce its Cognitive Complexity from 21 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 67 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 62 major
* ![MINOR][MINOR] 4 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [ColonyManager.java#L956](https://github.com/Minecolonies/minecolonies/blob/0753e0462f62a24d1f015b1d76f9ed0a04158d8c/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L956): Refactor this code to not nest more than 5 if/for/while/switch/try statements. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS134)
1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/0753e0462f62a24d1f015b1d76f9ed0a04158d8c/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/0753e0462f62a24d1f015b1d76f9ed0a04158d8c/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [WindowHutFarmer.java#L214](https://github.com/Minecolonies/minecolonies/blob/0753e0462f62a24d1f015b1d76f9ed0a04158d8c/src/main/java/com/minecolonies/coremod/client/gui/WindowHutFarmer.java#L214): Reduce this anonymous class number of lines from 43 to at most 40, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/0753e0462f62a24d1f015b1d76f9ed0a04158d8c/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [colony#](https://github.com/Minecolonies/minecolonies/blob/0753e0462f62a24d1f015b1d76f9ed0a04158d8c/src/main/java/com/minecolonies/coremod/colony#): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [ColonyManager.java#L55](https://github.com/Minecolonies/minecolonies/blob/0753e0462f62a24d1f015b1d76f9ed0a04158d8c/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L55): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 38 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [ColonyManager.java#L55](https://github.com/Minecolonies/minecolonies/blob/0753e0462f62a24d1f015b1d76f9ed0a04158d8c/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L55): class "ColonyManager" has 57 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [ColonyManager.java#L331](https://github.com/Minecolonies/minecolonies/blob/0753e0462f62a24d1f015b1d76f9ed0a04158d8c/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L331): Provide multiple methods instead of using "canDestroy" to determine which action to take. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2301)
1. ![MAJOR][MAJOR] [ColonyManager.java#L596](https://github.com/Minecolonies/minecolonies/blob/0753e0462f62a24d1f015b1d76f9ed0a04158d8c/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L596): Refactor this method to reduce its Cognitive Complexity from 21 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 67 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 62 major
* ![MINOR][MINOR] 4 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [ColonyManager.java#L956](https://github.com/Minecolonies/minecolonies/blob/5e1396e71a3821094f7c3020536aee2c10958f0c/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L956): Refactor this code to not nest more than 5 if/for/while/switch/try statements. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS134)
1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/5e1396e71a3821094f7c3020536aee2c10958f0c/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/5e1396e71a3821094f7c3020536aee2c10958f0c/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [WindowHutFarmer.java#L214](https://github.com/Minecolonies/minecolonies/blob/5e1396e71a3821094f7c3020536aee2c10958f0c/src/main/java/com/minecolonies/coremod/client/gui/WindowHutFarmer.java#L214): Reduce this anonymous class number of lines from 43 to at most 40, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/5e1396e71a3821094f7c3020536aee2c10958f0c/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [colony#](https://github.com/Minecolonies/minecolonies/blob/5e1396e71a3821094f7c3020536aee2c10958f0c/src/main/java/com/minecolonies/coremod/colony#): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [ColonyManager.java#L55](https://github.com/Minecolonies/minecolonies/blob/5e1396e71a3821094f7c3020536aee2c10958f0c/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L55): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 38 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [ColonyManager.java#L55](https://github.com/Minecolonies/minecolonies/blob/5e1396e71a3821094f7c3020536aee2c10958f0c/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L55): class "ColonyManager" has 57 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [ColonyManager.java#L331](https://github.com/Minecolonies/minecolonies/blob/5e1396e71a3821094f7c3020536aee2c10958f0c/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L331): Provide multiple methods instead of using "canDestroy" to determine which action to take. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2301)
1. ![MAJOR][MAJOR] [ColonyManager.java#L596](https://github.com/Minecolonies/minecolonies/blob/5e1396e71a3821094f7c3020536aee2c10958f0c/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L596): Refactor this method to reduce its Cognitive Complexity from 21 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 67 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 62 major
* ![MINOR][MINOR] 4 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [ColonyManager.java#L956](https://github.com/Minecolonies/minecolonies/blob/f0f08771a2c7880fa3b4a9d824931d2e16213b2a/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L956): Refactor this code to not nest more than 5 if/for/while/switch/try statements. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS134)
1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/f0f08771a2c7880fa3b4a9d824931d2e16213b2a/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/f0f08771a2c7880fa3b4a9d824931d2e16213b2a/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [WindowHutFarmer.java#L214](https://github.com/Minecolonies/minecolonies/blob/f0f08771a2c7880fa3b4a9d824931d2e16213b2a/src/main/java/com/minecolonies/coremod/client/gui/WindowHutFarmer.java#L214): Reduce this anonymous class number of lines from 43 to at most 40, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/f0f08771a2c7880fa3b4a9d824931d2e16213b2a/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [colony#](https://github.com/Minecolonies/minecolonies/blob/f0f08771a2c7880fa3b4a9d824931d2e16213b2a/src/main/java/com/minecolonies/coremod/colony#): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [ColonyManager.java#L55](https://github.com/Minecolonies/minecolonies/blob/f0f08771a2c7880fa3b4a9d824931d2e16213b2a/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L55): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 38 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [ColonyManager.java#L55](https://github.com/Minecolonies/minecolonies/blob/f0f08771a2c7880fa3b4a9d824931d2e16213b2a/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L55): class "ColonyManager" has 57 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [ColonyManager.java#L331](https://github.com/Minecolonies/minecolonies/blob/f0f08771a2c7880fa3b4a9d824931d2e16213b2a/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L331): Provide multiple methods instead of using "canDestroy" to determine which action to take. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2301)
1. ![MAJOR][MAJOR] [ColonyManager.java#L596](https://github.com/Minecolonies/minecolonies/blob/f0f08771a2c7880fa3b4a9d824931d2e16213b2a/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L596): Refactor this method to reduce its Cognitive Complexity from 21 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by marchermans — APPROVED** [review]

**Pull request merged**